Documentation/EtcdLease.md

```diff
@@ -0,0 +1,38 @@
+# Overview
+The EtcdLease implement interface `grant` and `revoke` to new a lease and revoke a lease to etcd.
+It can manage the leases to send keep alive request for leases.
+
+
+## `startKeepAliveService` function
+..* The function will start a backgroud scheduler `keepAliveSchedule` to send keep alive request for lease registered to EtcdLease Client.
+..* It will open a StreamObserver to etcd, send request to this StreamObserver can keep the lease alive
+..* It will new a ScheduledExecutorService to run `keepAliveExecutor` and `deadLineExecutor` periodically
+
+## `closeKeepAliveService` function
+..* This function will end the backgroud scheduler.
+..* It will close the StreamObserver to etcd
+
+## `keepAlive` function
+..* It will new a Lease object to keep the lease id and nextKeepAliveTime and DeadLine
+..* It will add the lease object to keepAlives map
+..* the backgroud scheduler will send keep alive request for the registered leases. 
```

Comment:
The background scheduler will send keep alive requests for the added lease when it approaches its nextKeepAliveTime.